### PR TITLE
Update Tab styles

### LINKED ISSUE #1662: Adjust and expand the Tab component
### Description

Currently, the Tab component is of a fixed width, which creates problems when used with longer text labels. The goal of this task is to:
- [ ] adjust the sizing of the Tab component so that its width is being driven by the overall width of the child elements inside and the left + right padding (to be specified by the designer),
- [ ] additionally, for the purpose of new NFT screens in the app, there's a need to **implement** a new variant of the Tab component that features a pill component nested inside for displaying numbers.

https://www.figma.com/file/Pf31tuYpozYmpq163U2ho8/Web-Components?node-id=110%3A18286

### Figma link

https://www.figma.com/file/Pf31tuYpozYmpq163U2ho8/Web-Components?node-id=110%3A18286

### How urgent this is?

This will block me soon

### LINKED ISSUE #1669: Tabs in My videos force horizontal overflow on mobile
**Describe the bug**

When there isn't enough space to display all the tabs in My videos view, the tabs don't properly hide but instead force the screen to overflow.

**Screenshots**
Screenshot 2021-11-27 at 01 03 42

(this is 400px wide screen)

### PR DESCRIPTION
**Pull request by rafalpawlow**
Closes #1662 
Fixes #1669 
https://www.figma.com/file/Pf31tuYpozYmpq163U2ho8/Web-Components?node-id=110%3A18286

### CODE REVIEW

File: src/components/Tabs/Tabs.tsx [hunk 1]
```diff
@@ -2,107 +2,118 @@ import { throttle } from 'lodash-es'
...
-  const [shadowsVisible, setShadowsVisible] = useState({
-    left: false,
-    right: true,
-  })
+export const Tabs: React.FC<TabsProps> = React.memo(
+  ({ tabs, onSelectTab, initialIndex = -1, selected: paramsSelected, className }) => {
+    const [_selected, setSelected] = useState(initialIndex)
+    const selected = paramsSelected ?? _selected
+    const [isContentOverflown, setIsContentOverflown] = useState(false)
+    const tabsRef = useRef<HTMLDivElement>(null)
```
Comment (drillprop) [inline src/components/Tabs/Tabs.tsx]: let's name this `tabsGroupRef` like name of the component

**Review by drillprop — CHANGES_REQUESTED** [review]
Could we also update the storybook for tabs to see variants with pills and badges?

**Comment by kdembler** [comment]
Ugh, I see that we didn't have the border in the component previously, it was always added in the view directly :/ Could we introduce this here? (will unfortunately require updating views where we use it)

Screenshot 2021-12-06 at 15 23 47

File: src/views/viewer/ChannelView/ChannelView.styles.ts [hunk 2]
```diff
@@ -131,22 +131,33 @@ export const PaginationContainer = styled.div`
...
-    border-bottom: solid 1px ${oldColors.gray[800]};
     gap: ${sizes(8)};
     grid-template: 1fr / auto 1fr 250px;
   }
 `
 
+export const UtilitiesContainer = styled.div`
+  display: grid;
+  gap: ${sizes(2)};
+  grid-template: 'search search search' auto 'sort sort sort' auto / 1fr 1fr;
```
Comment (drillprop) [inline src/views/viewer/ChannelView/ChannelView.styles.ts]: Sorry for bothering you :D, but I'm pretty sure, that you don't need to set this template and grid-areas.

I opened PR https://github.com/wyrdrender/atlas/pull/3

Let me know what you think.

Comment (rafalpawlow) [reply src/views/viewer/ChannelView/ChannelView.styles.ts]: Merged!

File: src/components/Badge/Badge.styles.ts [hunk 3]
```diff
@@ -16,6 +16,9 @@ export const smallBadgeStyles = css`
     text-transform: ${cVar('typographyDesktopT100TextTransform')};
     padding-top: 1px;
     text-align: center;
+    margin-left: ${sizes(2)};
```
Comment (kdembler) [inline src/components/Badge/Badge.styles.ts]: Do we really need the badge to have a `margin-left` all the time? Seems like this is only specific use-case

File: src/components/Pill/Pill.tsx [hunk 4]
```diff
@@ -7,7 +7,7 @@ import { PillProps } from './types'
 
 export const Pill: React.FC<PillProps> = ({ label, icon, iconPlacement = 'left', ...props }) => {
   return (
-    <StyledPill {...props} hasLabel={!!(label && label.length)} iconPlacement={iconPlacement}>
+    <StyledPill {...props} hasLabel={!!(label && label === 0)} iconPlacement={iconPlacement}>
```
Comment (kdembler) [inline src/components/Pill/Pill.tsx]: `hasLabel={label === 0}` seem counter-intuitive, what is this supposed to represent?

**Review by kdembler — CHANGES_REQUESTED** [review]
2 additional things:
1. The pill tabs don't look correct in storybook
Screenshot 2021-12-13 at 11 53 02
2. (I think) because of the `margin-left` on badge, the navigation breaks slightly:
Screenshot 2021-12-13 at 11 53 55

**Review by kdembler — APPROVED** [review]

**Pull request merged**